infra/helper.py

```diff
@@ -541,31 +545,39 @@ def download_corpus(project_name):
           file=sys.stderr)
     return False
 
-  fuzz_targets = _get_fuzz_targets(project_name)
-  corpus_dir = _get_corpus_dir(project_name)
+  if args.fuzz_target:
+    fuzz_targets = [args.fuzz_target]
+  else:
+    fuzz_targets = _get_fuzz_targets(args.project_name)
+
+  corpus_dir = _get_corpus_dir(args.project_name)
   if not os.path.exists(corpus_dir):
     os.makedirs(corpus_dir)
 
   def _download_for_single_target(fuzz_target):
     try:
-      _get_latest_corpus(project_name, fuzz_target, corpus_dir)
+      _get_latest_corpus(args.project_name, fuzz_target, corpus_dir)
       return True
     except Exception as e:
       print('ERROR: corpus download for %s failed: %s' % (fuzz_target, str(e)),
             file=sys.stderr)
       return False
 
-  print('Downloading corpus for %s project' % project_name)
+  print('Downloading corpus for %s project' % args.project_name)
   thread_pool = ThreadPool(multiprocessing.cpu_count())
   return not False in thread_pool.map(_download_for_single_target, fuzz_targets)
 
 
 def profile(args):
   """Generate code coverage using clang source based code coverage."""
+  if args.corpus_dir and not args.fuzz_target:
+    print('ERROR: --corpus-dir argument cannot be used without --fuzz-target',
+          file=sys.stderr)
+    return 1
   if not _check_project_exists(args.project_name):
     return 1
-  if not args.no_corpus_download:
-    if not download_corpus(args.project_name):
+  if not args.no_corpus_download and not args.corpus_dir:
+    if not download_corpus(args):
      return 1
```

Comment:
is the indent correct, seems off-by-one.

Reply:
Indeed! Done.